src/axom/sidre/spio/IOManager.cpp

```diff
@@ -463,13 +463,27 @@ void IOManager::loadExternalData(sidre::Group* datagroup,
  *
  *************************************************************************
  */
-void IOManager::createRootFile(const std::string& file_base,
-                               int num_files,
-                               const std::string& protocol,
-                               const std::string& tree_pattern)
+std::string IOManager::createRootFile(const std::string& root_base,
+                                      int num_files,
+                                      const std::string& protocol,
+                                      const std::string& tree_pattern)
 {
   conduit::Node n;
 
+  std::string suffix(".root");
+  std::string file_base(root_base);
+
+  size_t base_size = file_base.size();
+  size_t suffix_size = suffix.size();
+  if(base_size > suffix_size)
+  {
+    if(file_base.compare(base_size - suffix_size, suffix_size, suffix) == 0)
+    {
+      // file_base has ".root" suffix, so remove it.
+      file_base.erase(base_size - suffix_size);
+    }
+  }
```

Comment:
I think you can replace this new block with
```cpp
file_base = axom::utilities::string::removeSuffix(file_base, ".root");
```

See: 
https://github.com/LLNL/axom/blob/cab5877121dbc467a716237eef1128c861a4baba/src/axom/core/utilities/StringUtilities.hpp#L66-L79

Reply:
Thanks, I wasn't aware of this utility.